clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -12397,6 +12397,9 @@ def warn_attr_min_eq_max:  Warning<
 def err_hlsl_attribute_number_arguments_insufficient_shader_model: Error<
   "attribute %0 with %1 arguments requires shader model %2 or greater">;
 
+def ext_hlsl_auto_type_specifier : ExtWarn<
```

Comment:
Should this warning default to error?

Reply:
No, I think we should just allow and warn on using extensions, that matches generally how Clang handles them unless you pass `-pedantic-errors` which switches all the extension warnings to errors.